src/components/refactor/dialogs/two-windings-transformer-creation/tap-changer-pane-table/tap-changer-pane-table.js

```diff
@@ -0,0 +1,576 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import React, { useCallback, useEffect, useState } from 'react';
+import { useFieldArray, useFormContext, useWatch } from 'react-hook-form';
+import {
+    Checkbox,
+    Grid,
+    IconButton,
+    Table,
+    TableBody,
+    TableCell,
+    TableContainer,
+    TableHead,
+    TableRow,
+    Tooltip,
+} from '@mui/material';
+import AddchartIcon from '@mui/icons-material/Addchart';
+import DragIndicatorIcon from '@mui/icons-material/DragIndicator';
+import { DragDropContext, Draggable, Droppable } from 'react-beautiful-dnd';
+import Papa from 'papaparse';
+import { useIntl } from 'react-intl';
+import FieldErrorAlert from '../../../rhf-inputs/field-error-alert';
+import IntegerInput from '../../../rhf-inputs/integer-input';
+import { TableNumericalInput } from '../../../rhf-inputs/table-inputs/table-numerical-input';
+import TapChangerPaneButtons from './tap-changer-pane-buttons';
+import CheckboxInput from '../../../rhf-inputs/booleans/checkbox-input';
+import AddRowsDialog from './add-rows-dialog';
+import { CreateRuleDialog } from '../create-rule/create-rule-dialog';
+import { ImportRuleDialog } from '../import-rule-dialog';
+import {
+    HIGH_TAP_POSITION,
+    LOW_TAP_POSITION,
+    STEPS,
+    STEPS_TAP,
+    TAP_POSITION,
+} from 'components/refactor/utils/field-constants';
+import { MAX_TAP_NUMBER } from '../two-windings-transformer-creation-dialog';
+import PropTypes from 'prop-types';
+
+function MultiCheckbox({
+    associatedArrayName,
+    selectedPropName,
+    handleClickIfChecked,
+    handleClickIfUnchecked,
+}) {
+    const arrayToWatch = useWatch({
+        name: associatedArrayName,
+    });
+
+    const allRowSelected = arrayToWatch.every((row) => row[selectedPropName]);
+
+    return (
+        <Checkbox
+            checked={arrayToWatch.length > 0 && allRowSelected}
+            onChange={(event) => {
+                event.target.checked
+                    ? handleClickIfChecked()
+                    : handleClickIfUnchecked();
+            }}
+        />
+    );
+}
+
+function DefaultTableCell({ tapChanger, rowIndex, column }) {
+    const valueToWatch = useWatch({
+        name: `${tapChanger}.${STEPS}[${rowIndex}].${column.dataKey}`,
+    });
+    return <TableCell key={column.dataKey}>{valueToWatch}</TableCell>;
```

Comment:
I wanted to do this at first but it was lot of new code instead of juste this 3 lines.
https://react-hook-form.com/api/useform/watch
"This method will watch specified inputs and return their values. **It is useful to render input value** and for determining what to render by condition."
Why do you say it should only be for effect ?

Reply:
I didn't read this part of the doc. My opinion is that we should use "watch" only when the component should react to a field change. It makes the code easier to understand
This component is very small which makes it fine, but what should never happen is someone reading this code, and thinking you should use "useWatch" everytime you need to display readonlyInputs.

I think such a simple component would work :

const ReadOnlyInput = {name} => {
    const {value} = useController({name});

    return ({value});
}

Reply:
You're right it's not that much code and it's clearer after all. Done.